### ISSUE TITLE
Near interaction improvements

### ISSUE THREAD

**Issue by CraigFeldspar** [BabylonJS/Babylon.js#10471]
After trying out near interaction on 3D GUI elements, I found several improvement that I would like to discuss with you @RaananW @satyapoojasama :

+ Near grab requires a ray to be hitting the mesh to trigger when the user pinches his fingers. This is okay for bigger meshes, but for little meshes, like resize handles, i can be hard to get a hit. I feel like near grab, like near press, should pick meshes only from a distance hit test (what you implemented as `pickWithMesh`)

+ Near press is not implemented, it is not possible to press buttons just by getting your index finger close. Everything is already there though, we just need to trigger pointerDown and Up events.

+ There is a small bug where nearGrab always takes precedence on nearPick, even when no hit was found, or mesh doesn't have `isNearGrabbable` set to true (causing near pick to have a `hit === false`).

+ `pickWithMesh` is rather inefficient, we hit test a sphere mesh versus another meshes, causing a lot of triangle-triangle intersections. We can just perform distance-to-triangle tests instead.

I already started with implementing some of that because I need these improvements to test the new `NearMenu`, but I would be happy to get your opinion on these.

**Comment by RaananW** [comment]
Hi @CraigFeldspar , I won't find time in the near future to tackle those, and @satyapoojasama will probably not be able to help here as well ATM.
I noticed that you self-assigned yourself - awesome! If there are any issues let me know.

**Comment by CraigFeldspar** [comment]
Hi @RaananW, yep I'm on it ! Just wanted you guys to be aware of that if something comes to your mind in the meantime.

**Comment by satyapoojasama** [comment]
Hi @CraigFeldspar , thank you for testing near interaction thoroughly! I am fairly new to writing code in javascript and hand interactions in general which has been a challenge for me. I have switched teams recently and will not be able to work on this anymore. I appreciate you working on them.

**Comment by rickfromwork** [comment]
@CraigFeldspar Are you planning on updating any existing controls/behaviours/gizmos to respond to near interaction as part of these improvements? The linked PR is hard to parse at the moment, so I'm not sure if you've already done work towards that or not.

**Comment by CraigFeldspar** [comment]
@rickfromwork all points referenced above should are fixed by the linked PR. I will update the issue or create a new one with other small near interaction issues that I found out recently.
